internal/olm/client/client.go

```diff
@@ -280,18 +310,16 @@ func (c Client) printPodErrors(ctx context.Context, depSelectors *metav1.LabelSe
 		return fmt.Errorf("error getting Pods: %v", err)
 	}
 	for _, p := range podList.Items {
-		if p.Status.Phase != corev1.PodSucceeded {
-			for _, cs := range p.Status.ContainerStatuses {
-				if !cs.Ready {
-					podErrors[p.Name] = cs.State.Waiting.Message
-				}
+		for _, cs := range p.Status.ContainerStatuses {
+			if !cs.Ready {
+				podErr = append(podErr, deploymentError{
+					depName: p.Name + cs.Name,
+					issue:   cs.State.Waiting.Message,
```

Comment:
cannot it be nil?

Reply:
Which field are you talking about?

Reply:
`cs.State.Waiting.Message`

Reply:
if its empty, then nothing will get populated. We  can still inform user about the pod that has failed.

Reply:
if State or Waiting == nil it will cause panic.